internal/processors/cloudeventconvert/cloudeventconvert.go

```diff
@@ -190,17 +263,37 @@ func setMetaData(hdr *cloudevent.CloudEventHeader, msg *service.Message) {
 	msg.MetaSetMut(cloudEventProducerKey, hdr.Producer)
 	msg.MetaSetMut(cloudEventSubjectKey, hdr.Subject)
 	msg.MetaSetMut(cloudEventIDKey, hdr.ID)
+	msg.MetaSetMut(httpinputserver.DIMOCloudEventSource, hdr.Source)
+}
+
+func processAttestation(msgBytes []byte) (*cloudevent.CloudEvent[json.RawMessage], error) {
+	var event cloudevent.CloudEvent[json.RawMessage]
+	if err := json.Unmarshal(msgBytes, &event); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal attestation cloud event: %w", err)
+	}
+
+	if processors.IsFutureTimestamp(event.Time) {
+		return nil, fmt.Errorf("event timestamp %v exceeds valid range", event.Time)
+	}
+
+	if _, err := cloudevent.DecodeNFTDID(event.Subject); err != nil {
+		return nil, fmt.Errorf("invalid attestation subject format: %w", err)
+	}
+
+	event.Type = cloudevent.TypeAttestation
+	return &event, nil
 }
 
 func isValidCloudEventHeader(eventHdr *cloudevent.CloudEventHeader) bool {
 	if _, err := cloudevent.DecodeNFTDID(eventHdr.Subject); err != nil {
 		return false
 	}
+
 	if _, err := cloudevent.DecodeNFTDID(eventHdr.Producer); err != nil {
-		return false
-	}
-	if !common.IsHexAddress(eventHdr.Source) {
-		return false
+		if _, err = cloudevent.DecodeEthrDID(eventHdr.Producer); err != nil {
```

Comment:
nitpick: Could we leave a comment here? This looked like a typo at first glance.